lib/roby/queries/code_error_matcher.rb

```diff
@@ -11,7 +11,7 @@ class CodeErrorMatcher < LocalizedErrorMatcher
 
             def initialize
                 super
-                @ruby_exception_class = ::Exception
+                @ruby_exception_class = Any
```

Comment:
why isnt matching things other than exceptions an issue?

Reply:
Because there is the "no exception at all" case. Which is represented by `@ruby_exception_class` being nil in the code. It is common for EmissionFailed (that's how I found out about the issue).

Essentially, I was expecting that

```
expect_execution { .... }.to { have_error_matching EmissionFailed }
```

would work in all cases, but it did not.

Reply:
I meant that `@ruby_exception_class` is accepting any Object, not only exceptions or nil. Isnt that an issue? I couldnt find any places that could have issues with this in CodeErrorMatcher/LocalizedErrorMatcher, but its feeling wrong.

Reply:
@ruby_exception_class is a matcher, actually. It is only used with `#===` to verify whether or no it matches the exception object. That's what Any works (Any's `===` returns true always)

Reply:
(can see that in the documentation of with_ruby_exception)